internal/internal_task_pollers.go

```diff
@@ -964,32 +944,23 @@ func reportActivityCompleteByID(ctx context.Context, service workflowservice.Wor
 	var reportErr error
 	switch request := request.(type) {
 	case *workflowservice.RespondActivityTaskCanceledByIdRequest:
-		reportErr = backoff.Retry(ctx,
-			func() error {
-				grpcCtx, cancel := newGRPCContext(ctx, grpcMetricsScope(rpcScope))
-				defer cancel()
-
-				_, err := service.RespondActivityTaskCanceledById(grpcCtx, request)
-				return err
-			}, createDynamicServiceRetryPolicy(ctx), isServiceTransientError)
+		grpcCtx, cancel := newGRPCContext(ctx, grpcMetricsScope(rpcScope),
+			grpcContextValue(retry.ConfigKey, createDynamicServiceRetryPolicy(ctx).IntoGrpcRetryConfig()))
+		defer cancel()
+		_, err := service.RespondActivityTaskCanceledById(grpcCtx, request)
+		reportErr = err
 	case *workflowservice.RespondActivityTaskFailedByIdRequest:
-		reportErr = backoff.Retry(ctx,
-			func() error {
-				grpcCtx, cancel := newGRPCContext(ctx, grpcMetricsScope(rpcScope))
-				defer cancel()
-
-				_, err := service.RespondActivityTaskFailedById(grpcCtx, request)
-				return err
-			}, createDynamicServiceRetryPolicy(ctx), isServiceTransientError)
+		grpcCtx, cancel := newGRPCContext(ctx, grpcMetricsScope(rpcScope),
+			grpcContextValue(retry.ConfigKey, createDynamicServiceRetryPolicy(ctx).IntoGrpcRetryConfig()))
```

Comment:
This line gets repeated a lot and could probably be factored out into a simpler one-function call.

Reply:
It's actually repeated a lot in slightly different variations. What we can do is extract `grpcContextValue(retry.ConfigKey, createDynamicServiceRetryPolicy(ctx).IntoGrpcRetryConfig())` into a helper function and call it something like defaultGrpcRetryContextValue`

Reply:
Yeah, I think that would be useful